elasticsearch/plugins.sls

```diff
@@ -1,17 +1,21 @@
 include:
   - elasticsearch.pkg
 
-{%- if salt['pillar.get']('elasticsearch:plugins', {}) %}
+{%- set plugins_pillar = salt['pillar.get']('elasticsearch:plugins', {}) %}
 
-{% for name, repo in elasticsearch.plugins.iteritems() %}
+{%- set major_version = salt['pillar.get']('elasticsearch:major_version', 2) %}
+
+{% if major_version == 5 %}
+  {%- set plugin_bin = 'elasticsearch-plugin' %}
+{% else %}
+  {%- set plugin_bin = 'plugin' %}
+{% endif %}
+
+{% for name, repo in plugins_pillar.items() %}
 elasticsearch-{{ name }}:
   cmd.run:
-    - name: /usr/share/elasticsearch/bin/plugin install {{ repo }}
+    - name: /usr/share/elasticsearch/bin/{{ plugin_bin }} install -b {{ repo }}
```

Comment:
``-b,--batch                   Enable batch mode explicitly, automatic confirmation of security permissions``

And since this is not interactive installation it makes sense.